modules/editor-service/src/main/java/org/opencastproject/editor/EditorServiceImpl.java

```diff
@@ -786,6 +867,12 @@ public void setEditData(String mediaPackageId, EditingData editingData) throws E
       errorExit("Unable to add SMIL cutting catalog to archive", mediaPackageId, ErrorStatus.UNKNOWN, e);
     }
 
+    try {
+      addSubtitleTrack(mediaPackage, editingData.getSubtitles());
+    } catch (IOException e) {
```

Comment:
} catch (IOException | NullPointerException e) {

(Should you decided to heed my suggestion about throwing NullPointerException)

Reply:
Caught the NullPointerException in a different clause in [7a78ba0](https://github.com/opencast/opencast/pull/3604/commits/7a78ba0efafe259f6663aaf6efaf818e59d61f00), because it is perfectly valid for the frontend to not send any subtitles.